vcluster/deploy/environment/eks.mdx

```diff
@@ -0,0 +1,210 @@
+  :::note
+  [Upgrade](https://github.com/eksctl-io/eksctl?tab=readme-ov-file#installation) `eksctl` to the latest version to ensure latest Kubernetes version is
+  deployed.
+  :::
+
+## Create EKS cluster
+
+Start by creating EKS cluster using `eksctl`. This command creates a file named
+`cluster.yaml` with the required settings. Adjust the cluster name, region, and instance type as needed.
+
+```bash title="Cluster configuration file"
+CLUSTER_NAME=vcluster-demo
+REGION=eu-central-1
+INSTANCE_TYPE=t3.medium
+
+cat << EOF > cluster.yaml
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+metadata:
+ name: ${CLUSTER_NAME}
+ region: ${REGION}
+iam:
+  withOIDC: true
+nodeGroups:
+ - name: ng-1
+   instanceType: ${INSTANCE_TYPE}
+   desiredCapacity: 2
+   volumeSize: 80
+
+addons:
+- name: aws-ebs-csi-driver
+  wellKnownPolicies:      # this setting adds IAM and service account
+    ebsCSIController: true
+EOF
+```
+
+The file defines a cluster with two `t3.medium` instances in the `eu-central-1`
+region. The `aws-ebs-csi-driver` add-on is enabled to ensure EBS storage
+provisioning which is required by `vCluster` to store about virtual clusters.
+
+Create the cluster by running:
+
+```bash title="Create EKS cluster"
+eksctl create cluster -f cluster.yaml
+```
+
+<KubeconfigUpdate />
+
+This process typically takes about 15-20 minutes.
+
+### Verify the cluster creation
+
+Verify the installation by checking if the CSI driver pods are running:
+
+```bash title="Verify CSI driver installation"
+kubectl get pods -n kube-system -l app.kubernetes.io/name=aws-ebs-csi-driver
+```
+
+Expected output should look similar to:
+```
+NAME                                 READY   STATUS    RESTARTS   AGE
+ebs-csi-controller-794b4448b-fhjxr   6/6     Running   0          2m14s
+ebs-csi-controller-794b4448b-j94g5   6/6     Running   0          2m14s
+ebs-csi-node-crz7p                   3/3     Running   0          2m14s
+ebs-csi-node-jg8n8                   3/3     Running   0          2m14s
+```
+
+### Configure storage class
+
+`vCluster` requires a default StorageClass for its persistent volumes. EKS provides the `gp2` StorageClass by default, but `gp3` is recommended for better performance and cost efficiency. Create a new StorageClass:
+
+```bash title="gp3 StorageClass configuration"
+cat <<EOF | kubectl apply -f -
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: gp3
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
+provisioner: ebs.csi.aws.com
+parameters:
+  type: gp3
+  encrypted: "true"
+volumeBindingMode: WaitForFirstConsumer
+allowVolumeExpansion: true
+EOF
+```
+
+Remove the default status from the `gp2` StorageClass:
+
+```bash title="Remove default status from gp2 StorageClass"
+kubectl patch storageclass gp2 -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"false"}}}'
+```
+
+### Configure workload identity
```

Comment:
Is this a requirement or nice to have if you want to do pod identity?

Reply:
I'm not sure if this is a requirement, but most platform engineers would want this.